app/reducers/optional.js

```diff
@@ -1,7 +0,0 @@
-//this is a simple pass thru that offers an extension point
```

Comment:
Maybe I'm still missing something but the solution I'm proposing offers:

- works immediately after install ... reducer just returns the initial state but it works without error
- easily understandable way to add `initialState` which can be very useful; defaults to empty object
- allows a simple `ember generate` / `ember destroy` metaphor for extending the reducers

I did find your documentation around `optional.js` and I think this should be deprecated for a variety of reasons:

- it is not in any part a standard part of Redux
- it's naming is confusing and the fact that you have a non-reducer function in the `/reducers` directory is a bad idea
- you're own admission that you had added it in the first place because you hadn't fully grokked middleware when you added it

Functionally the combination of _initial state_ and _middleware_ should meet any use case that historically used _optional_ but as you point out removing it outright can only happen on a major version but I'd suggest making this section a footnote in the `config` section of your documentation and logging a deprecation when it's used.